onnx/defs/shape_inference.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "data_type_utils.h"
+
+namespace ONNX_NAMESPACE {
+
+struct InferenceContext {
+  virtual const AttributeProto* getAttribute(const std::string& name) const = 0;
+  virtual size_t getNumInputTypes() const = 0;
+  virtual const TypeProto_Tensor* getInputType(size_t index) const = 0;
+  virtual size_t getNumOutputTypes() const = 0;
+  virtual TypeProto_Tensor* getOutputType(size_t index) = 0;
+  virtual ~InferenceContext() {}
+};
+
+typedef void (*InferenceFunction)(InferenceContext&);
+
+template<typename T> inline google::protobuf::RepeatedField<T> retrieveValues(const AttributeProto* attr);
+template<> inline google::protobuf::RepeatedField<int64_t> retrieveValues(const AttributeProto* attr) { return attr->ints(); }
+
+template<typename T>
+inline bool getRepeatedAttribute(InferenceContext& ctx,
```

Comment:
maybe make it part of the InferenceContext for consistency?

Reply:
I will do this iff @linkerzhang is ok with it

Reply:
it seems to me that they're helper functions, let's keep them outside of the interface for now. Thank you very much!